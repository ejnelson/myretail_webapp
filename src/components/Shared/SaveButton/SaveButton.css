.saveButton {
  width: 31%;
  border: none;
  background-color: #eee;
  color: #aaa;
  cursor: pointer;
  border-radius: 3px;
  margin: 1%;
  padding: 4px;
  outline: none;
}
