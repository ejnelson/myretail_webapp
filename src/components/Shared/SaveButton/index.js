import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './SaveButton.css';

export default class SaveButton extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { buttonText, action } = this.props;
    return (
      <button className="saveButton" onClick={action}>
        {buttonText}
      </button>
    );
  }
}
