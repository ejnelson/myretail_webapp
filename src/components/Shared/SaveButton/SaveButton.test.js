import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import SaveButton from './index';

describe('SaveButton', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SaveButton />, div);
  });
  it('renders correctly', () => {
    const tree = renderer.create(<SaveButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('handles props', () => {
    const saveButton = mount(
      <SaveButton action={console.log('test')} buttonText="test" />
    );
    saveButton.find('button').simulate('click');
    // TODO expect click to fire
    expect(saveButton.text()).toEqual('test');
  });
});
