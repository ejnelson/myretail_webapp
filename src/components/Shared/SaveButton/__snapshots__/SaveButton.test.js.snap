// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveButton renders correctly 1`] = `
<button
  className="saveButton"
  onClick={undefined}
/>
`;
